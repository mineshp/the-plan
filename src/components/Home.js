import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import {
    Button,
    Container,
    Divider,
    Grid,
    Header,
    Segment
} from 'semantic-ui-react';

export default class HomepageLayout extends Component {
    constructor() {
        super();
        this.state = {};
    }

    // eslint-disable-next-line class-methods-use-this
    render() {
        return (
            <div className="content-body">
                <Segment
                    textAlign="center"
                    className="home-segment"
                    vertical
                >
                    <Container text>
                        <Header
                            as="h2"
                            content="Time to make plans..."
                            className="home-mainheader"
                        />
                        <Header
                            as="h3"
                            content="Stars can't shine without darkness."
                            className="home-header"
                        />
                        <Divider inverted section />
                        <Grid stackable divided="vertically" centered>
                            <Grid.Row columns={3}>
                                <Grid.Column>
                                    <Button
                                        as={Link}
                                        color="green"
                                        icon="list layout"
                                        content="View lists"
                                        to="/list/all"
                                    />
                                </Grid.Column>
                                <Grid.Column>
                                    <Button
                                        as={Link}
                                        color="pink"
                                        icon="edit"
                                        content="Create lists"
                                        to="/list/update"
                                    />
                                </Grid.Column>
                                <Grid.Column>
                                    <Button
                                        as={Link}
                                        color="blue"
                                        icon="cubes"
                                        content="View Projects"
                                        to="/project/all"
                                    />
                                </Grid.Column>
                            </Grid.Row>
                        </Grid>
                        <div />
                    </Container>
                </Segment>
            </div>
        );
    }
}
