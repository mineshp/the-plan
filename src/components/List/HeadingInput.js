import React from 'react';
import PropTypes from 'prop-types';
import { Button, Divider, Form, Grid, Input } from 'semantic-ui-react';

const HeadingInput = ({ headings, handleAddColumn, handleRemoveColumn, handleHeaderInputChange }) => {
    const headerRows = [];
    headings.map((heading) => {
        const headingID = heading.id; // eslint-disable-line no-underscore-dangle
        return (
            headerRows.push(
                <Grid.Row key={headingID}>
                    <Grid.Column>
                        <Form.Field>
                            <Input
                                placeholder="Header Name..."
                                name={heading.name}
                                id={headingID}
                                key={headingID}
                                defaultValue={heading.name}
                                onChange={handleHeaderInputChange}
                            />
                        </Form.Field>
                    </Grid.Column>
                    <Grid.Column>
                        <Button
                            icon="trash"
                            content="Remove"
                            labelPosition="left"
                            id={headingID}
                            color="pink"
                            type="button"
                            onClick={handleRemoveColumn}
                            floated="left"
                        />
                    </Grid.Column>
                </Grid.Row>
            )
        );
    });

    return (
        <div>
            <Grid columns={2} divided>
                <Divider horizontal>ASSIGN LIST HEADINGS</Divider>
                {headerRows}
                <Divider />
                <Button fluid type="button" color="blue" onClick={handleAddColumn}>Add Column</Button>
            </Grid>
        </div>
    );
};

HeadingInput.propTypes = {
    headings: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string
        }).isRequired
    ).isRequired,
    handleAddColumn: PropTypes.func.isRequired,
    handleRemoveColumn: PropTypes.func.isRequired,
    handleHeaderInputChange: PropTypes.func.isRequired
};

export default HeadingInput;
