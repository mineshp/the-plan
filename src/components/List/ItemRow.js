import React from 'react';
import PropTypes from 'prop-types';
import { Button, Table } from 'semantic-ui-react';


const ItemRow = ({ data, listID }) => (
    <Table.Row>
        <Table.Cell>{data.name}</Table.Cell>
        <Table.Cell>{data.createdDate}</Table.Cell>
        <Table.Cell>{data.updatedDate}</Table.Cell>
        <Table.Cell>{data.notes}</Table.Cell>
        <Table.Cell>{data.owner}</Table.Cell>
        <Table.Cell><Button as='a' color='teal' href={`/list/${listID}/item/${data.id}/delete`}>Delete</Button></Table.Cell>
    </Table.Row>
);

ItemRow.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    createdDate: PropTypes.string.isRequired,
    updatedDate: PropTypes.string.isRequired
    }).isRequired
}

export default ItemRow;