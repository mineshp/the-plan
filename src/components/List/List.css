.ui.input.stdInput {
    width: 90% !important;
}

.ui.formSpacing {
    margin-top: 40px;
}