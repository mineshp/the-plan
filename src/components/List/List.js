import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'semantic-ui-react';
import DisplayMessage from '../Shared/DisplayMessage';

const Lists = ({
    errors,
    rows
}) => (
    <div className="List main">
        {
            (errors && errors.error) &&
                <DisplayMessage status={errors} />
        }
        <Table celled striped>
            <Table.Header>
                <Table.Row>
                    <Table.HeaderCell>Name</Table.HeaderCell>
                    <Table.HeaderCell>Created</Table.HeaderCell>
                    <Table.HeaderCell>Last Updated</Table.HeaderCell>
                    <Table.HeaderCell>Projects</Table.HeaderCell>
                    <Table.HeaderCell>Actions</Table.HeaderCell>
                </Table.Row>
            </Table.Header>

            <Table.Body>
                {rows}
            </Table.Body>
        </Table>
    </div>
);

Lists.propTypes = {
    errors: PropTypes.shape({}),
    rows: PropTypes.arrayOf(
        PropTypes.shape({})
    )
};

Lists.defaultProps = {
    errors: null,
    rows: []
};

export default Lists;
