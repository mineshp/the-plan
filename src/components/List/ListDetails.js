import React from 'react';
import PropTypes from 'prop-types';
import Table from 'semantic-ui-react/dist/commonjs/collections/Table';
import { formatDate } from '../../helpers/validators/common';

const ListDetails = ({ list }) => {
    const owner = list.owner ? list.owner : 'unknown';
    return (
        <Table>
            <Table.Body>
                <Table.Row>
                    <Table.Cell className="details-header">Owner:</Table.Cell>
                    <Table.Cell>{owner}</Table.Cell>
                    <Table.Cell className="details-header">Created Date:</Table.Cell>
                    <Table.Cell>{formatDate(list.createdDate)}</Table.Cell>
                    <Table.Cell className="details-header">Last Updated:</Table.Cell>
                    <Table.Cell>{formatDate(list.updatedDate)}</Table.Cell>
                </Table.Row>
            </Table.Body>
        </Table>
    );
};

ListDetails.propTypes = {
    list: PropTypes.shape({}).isRequired
};

export default ListDetails;
