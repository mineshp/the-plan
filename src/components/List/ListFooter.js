import React from 'react';
import PropTypes from 'prop-types';
import { Table, Button, Icon } from 'semantic-ui-react';

const ListFooter = ({ numColumns, handleAddItem, handleSubmit }) => (
    <Table.Footer fullWidth>
        <Table.Row>
            <Table.HeaderCell colSpan={numColumns}>
                <Button floated="left" color="blue" icon labelPosition="left" size="small" onClick={handleAddItem}>
                    <Icon name="plus" /> Add Item
                </Button>
                <Button
                    floated="right"
                    icon
                    labelPosition="left"
                    color="green"
                    size="small"
                    onClick={handleSubmit}
                >
                    <Icon name="save" /> Save
                </Button>
            </Table.HeaderCell>
        </Table.Row>
    </Table.Footer>
);

ListFooter.propTypes = {
    numColumns: PropTypes.number.isRequired,
    handleAddItem: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired
};

export default ListFooter;
