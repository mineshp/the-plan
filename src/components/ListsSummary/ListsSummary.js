import React from 'react';
import PropTypes from 'prop-types';
import { Container, Header, Table } from 'semantic-ui-react';

const ListsSummary = ({ retrieveListBy, rows }) => (
    <div className="List main content-body">
        <Container>
            <Header as="h1" className="firstletter-capitalise">{`Lists for ${retrieveListBy}.`}</Header>
            <Table striped>
                <Table.Header>
                    <Table.Row>
                        <Table.HeaderCell>Name</Table.HeaderCell>
                        <Table.HeaderCell>Created</Table.HeaderCell>
                        <Table.HeaderCell>Last Updated</Table.HeaderCell>
                        <Table.HeaderCell>Projects</Table.HeaderCell>
                        <Table.HeaderCell>Actions</Table.HeaderCell>
                    </Table.Row>
                </Table.Header>

                <Table.Body>
                    {rows}
                </Table.Body>
            </Table>
        </Container>
    </div>
);

ListsSummary.propTypes = {
    rows: PropTypes.arrayOf(
        PropTypes.shape({})
    ),
    retrieveListBy: PropTypes.string.isRequired
};

ListsSummary.defaultProps = {
    rows: []
};

export default ListsSummary;
