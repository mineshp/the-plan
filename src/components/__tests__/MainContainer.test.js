import React from 'react';
import { shallow } from 'enzyme';
import { Route } from 'react-router-dom';
import Home from '../Home';
import { ManageListConnectedComponent } from '../../HOC/List/ManageList';
import { ManageListSummaryConnectedComponent } from '../../HOC/ListsSummary/ManageListSummary';
import { UpdateProjectConnectedComponent } from '../../HOC/Project/Update';
import { ManageProjectConnectedComponent } from '../../HOC/Project/ManageProject';
import List from '../App';
import MainContainerRoutes from '../MainContainer';


describe('MainContainer Routes', () => {
    describe('Home Route', () => {
        it('validates path / renders Home component', () => {
            const tree = shallow(<MainContainerRoutes />);
            const attachedComponent =
                tree.findWhere((n) => n.prop('path') === '/').props().component;

            expect(attachedComponent).toBe(Home);
        });
    });

    describe('List Routes', () => {
        it('validates path /list/all renders ManageListSummary component', () => {
            const tree = shallow(<MainContainerRoutes />);
            const attachedComponent =
                tree.findWhere((n) => n.prop('path') === '/list/all').props().component;

            expect(attachedComponent).toBe(ManageListSummaryConnectedComponent);
        });

        it('validates path /list/1/view renders ViewList component', () => {
            const tree = shallow(<MainContainerRoutes />);
            const attachedComponent =
                tree.findWhere((n) => n.prop('path') === '/list/view/:id').props().component;

            expect(attachedComponent).toBe(ManageListConnectedComponent);
        });

        it('validates path /list renders List component', () => {
            const tree = shallow(<MainContainerRoutes />);
            const attachedComponent =
                tree.findWhere((n) => n.prop('path') === '/list').props().component;

            expect(attachedComponent).toBe(List);
        });
    });

    describe('Project Routes', () => {
        it('validates path /project/all renders ManageProject component', () => {
            const tree = shallow(<MainContainerRoutes />);
            const attachedComponent =
                tree.findWhere((n) => n.prop('path') === '/project/all').props().component;

            expect(attachedComponent).toBe(ManageProjectConnectedComponent);
        });

        it('validates path /project/update renders UpdateProjectConnectedComponent component for create', () => {
            const tree = shallow(<MainContainerRoutes />);
            const attachedComponent =
                tree.findWhere((n) => n.prop('path') === '/project/update').props().component;

            expect(attachedComponent).toBe(UpdateProjectConnectedComponent);
        });

        it('validates path /project/update/:id renders UpdateProjectConnectedComponent component for update', () => {
            const tree = shallow(<MainContainerRoutes />);
            const attachedComponent =
                tree.findWhere((n) => n.prop('path') === '/project/update/:id').props().component;

            expect(attachedComponent).toBe(UpdateProjectConnectedComponent);
        });
    });

    describe('All Routes', () => {
        it('Validates all routes', () => {
            const component = shallow(<MainContainerRoutes />);

            expect(component).toMatchSnapshot();
        });
    });

    describe('No Route', () => {
        xit('returns message if no match found for route', () => {
            const tree = shallow(<MainContainerRoutes />);

            const NoMatchRoute = tree.find(Route).last();
            console.log(NoMatchRoute.props().component);
            // expect(NoMatchRoute.props().component).toBe(NoMatchFunc);
            // expect(NoMatchRoute.contains(<div>No match for</div>)).toBe(true)
        });
    });
});
