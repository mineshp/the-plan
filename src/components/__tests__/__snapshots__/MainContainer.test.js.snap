// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer Routes All Routes Validates all routes 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Switch>
    <Route
        component={[Function]}
        exact={true}
        path="/"
    />
    <Route
        component={[Function]}
        path="/list/all"
    />
    <Route
        component={[Function]}
        path="/list/:id/view"
    />
    <Route
        component={[Function]}
        path="/list"
    />
    <Route
        component={[Function]}
        path="/project/all"
    />
    <Route
        component={[Function]}
        path="/project/update/:id"
    />
    <Route
        component={[Function]}
        path="/project/update"
    />
</Switch>,
  "nodes": Array [
    <Switch>
      <Route
            component={[Function]}
            exact={true}
            path="/"
      />
      <Route
            component={[Function]}
            path="/list/all"
      />
      <Route
            component={[Function]}
            path="/list/:id/view"
      />
      <Route
            component={[Function]}
            path="/list"
      />
      <Route
            component={[Function]}
            path="/project/all"
      />
      <Route
            component={[Function]}
            path="/project/update/:id"
      />
      <Route
            component={[Function]}
            path="/project/update"
      />
</Switch>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Main />,
      "_debugID": 15,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 8,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/list/all"
          />
          <Route
                    component={[Function]}
                    path="/list/:id/view"
          />
          <Route
                    component={[Function]}
                    path="/list"
          />
          <Route
                    component={[Function]}
                    path="/project/all"
          />
          <Route
                    component={[Function]}
                    path="/project/update/:id"
          />
          <Route
                    component={[Function]}
                    path="/project/update"
          />
</Switch>,
        "_debugID": 16,
        "_renderedOutput": <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/list/all"
          />
          <Route
                    component={[Function]}
                    path="/list/:id/view"
          />
          <Route
                    component={[Function]}
                    path="/list"
          />
          <Route
                    component={[Function]}
                    path="/project/all"
          />
          <Route
                    component={[Function]}
                    path="/project/update/:id"
          />
          <Route
                    component={[Function]}
                    path="/project/update"
          />
</Switch>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Main />,
}
`;
